influunt-api/app/controllers/ControladoresController.java

```diff
@@ -96,9 +105,49 @@
         }
     }
 
+
+    @Transactional
+    public CompletionStage<Result> edit(String id) {
+        if (getUsuario() == null) {
+            return CompletableFuture.completedFuture(unauthorized(Json.toJson(Arrays.asList(new Erro("clonar", "usuário não econtrado", "")))));
+        }
+
+        Controlador controlador = Controlador.find.byId(UUID.fromString(id));
+
+
+        if (controlador.getStatusControlador() != StatusControlador.ATIVO
+                && controlador.getStatusControlador() != StatusControlador.EM_CONFIGURACAO
+                && controlador.getStatusControlador() != StatusControlador.EM_EDICAO) {
+
+            return CompletableFuture.completedFuture(status(UNPROCESSABLE_ENTITY, Json.toJson(Arrays.asList(new Erro("clonar", "não é possível editar controlador", "")))));
+        }
+
+        if (controlador.getStatusControlador().equals(StatusControlador.EM_EDICAO) && !usuarioPodeEditarControlador(controlador, getUsuario())) {
+            return CompletableFuture.completedFuture(status(UNPROCESSABLE_ENTITY, Json.toJson(Arrays.asList(new Erro("editar", "usuário diferente do que está editando controlador!", "")))));
+        }
+
+        if (controlador.getStatusControlador() == StatusControlador.ATIVO) {
+            Controlador controladorEdicao = new ControladorUtil().provider(provider).deepClone(controlador);
+            VersaoControlador novaVersao = new VersaoControlador(controlador, controladorEdicao, getUsuario());
+            novaVersao.setDescricao("Controlador clonado pelo usuário: " + getUsuario().getNome());
+            novaVersao.save();
+
+            controlador.setStatusControlador(StatusControlador.CLONADO);
+            controlador.save();
+
+            return CompletableFuture.completedFuture(ok(new ControladorCustomSerializer().getControladorJson(controladorEdicao)));
+        }
+
+        if (controlador == null) {
+            return CompletableFuture.completedFuture(notFound());
```

Comment:
Se `controlador` for `null` vai dar NPE em todas as chamadas para `getStatusControlador()`, essa verificação tem de ser feita logo após buscar o controlador no banco.